Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -759,8 +759,17 @@ def network_shutdown(self):
             self.threadpool.cancel_all_pending_tasks()
             self.threadpool = None
 
+    def save_download_pstate_callback(self, resume_data, infohash, pstate):
+        pstate.set('state', 'engineresumedata', resume_data)
+        self.save_download_pstate(infohash, pstate)
```

Comment:
if `session` already killed, maybe yes. Otherwise, I don't think there's a problem with that.